src/admin/resolvers/queries/CollectionPartner.integration.ts

```diff
@@ -166,7 +169,41 @@ describe('queries: CollectionPartner', () => {
         variables: { id: 'invalid-id' },
       });
 
-      expect(data).toBeNull();
+      expect(data).to.be.null;
+    });
+  });
+
+  describe('getCollectionPartnerAssociation query', () => {
+    it('should get an association by its externalId', async () => {
+      const association = await createCollectionPartnerAssociationHelper(db, {
+        type: CollectionPartnershipType.PARTNERED,
+      });
+
+      const {
+        data: { getCollectionPartnerAssociation: data },
+      } = await server.executeOperation({
+        query: GET_COLLECTION_PARTNER_ASSOCIATION,
+        variables: { externalId: association.externalId },
+      });
+
+      expect(data).not.to.be.null;
+      expect(data.type).to.equal(CollectionPartnershipType.PARTNERED);
+      expect(data.partner).not.to.be.null;
+    });
+
+    it('should fail on an invalid externalId', async () => {
+      await createCollectionPartnerAssociationHelper(db, {
+        type: CollectionPartnershipType.PARTNERED,
+      });
+
+      const {
+        data: { getCollectionPartnerAssociation: data },
+      } = await server.executeOperation({
+        query: GET_COLLECTION_PARTNER_ASSOCIATION,
+        variables: { externalId: 'invalid-id' },
+      });
+
+      expect(data).to.be.null;
```

Comment:
should we add a check for `errors` in the returned response?

Reply:
you would think! i started this route, but then found out the `errors` array is never populated for this scenario. the `TODO` i added in another file in this PR addresses this issue. i also made a ticket to follow up: https://getpocket.atlassian.net/browse/BACK-1399

Reply:
fwiw this test is mimicking prior test behavior, so we aren't _losing_ any coverage.